crates/iceberg-catalog/src/api/management/v1/warehouse/mod.rs

```diff
@@ -377,6 +404,31 @@ pub trait Service<C: Catalog, A: Authorizer, S: SecretStore> {
         Ok(warehouses.into())
     }
 
+    async fn get_warehouse_statistics(
+        warehouse_id: WarehouseIdent,
+        query: GetWarehouseStatisticsQuery,
+        context: ApiContext<State<A, C, S>>,
+        request_metadata: RequestMetadata,
+    ) -> Result<WarehouseStatsResponse> {
+        // ------------------- AuthZ -------------------
+        let authorizer = context.v1_state.authz;
+        authorizer
+            .require_warehouse_action(
+                &request_metadata,
+                warehouse_id,
+                &CatalogWarehouseAction::CanGetMetadata,
```

Comment:
On second thought, I am not sure if guests, that have permissions only via inheritance, should be able to see stats. Lets leave it for now, but we might want to restrict this later if stats reveal more sensitive information about warehouses - i.e. occurances of table properties etc.

For those more sensitive stats we should introduce a new action that requires direct assignment to the warehouse or entities above - and can not "bubble up". 
We could use `describe` for it.